src/App.js

```diff
@@ -0,0 +1,254 @@
+
+import "./App.css";
+import { useState } from "react";
+import Papa from "papaparse";
+
+const Header = (props) => {
+  return (
+    <header>
+      <h1>{props.title}</h1>
+      <p>{props.message}</p>
+    </header>
+  );
+}
+
+function FileUpload(props) {
+  return(
+    <div>
+      <p>File name: {props.fileName}</p>
+      <p>File type: {props.fileType}</p>
+      <p>Size in bytes: {props.fileSize}</p>
+      <p>
+        Last modified date:{' '}
+        {props.fileDate}
+      </p>
+    </div>
+   )
+}
+
+const Instructions = (props) => {
+  return (
+    <div className='image-comparison-tool'>
+      <h2>Upload a CSV file to start comparing images</h2>
+      <p>This tool will display a side-by-side snapshot of two URLs from CSV.</p>
+      <h3>How to use</h3>
+      <ol>
+        <li>Prepare your CSV file with the following columns: <code>oldUrl</code>, <code>newUrl</code>, <code>status</code>.</li>
+        <li>Upload your CSV file.</li>
+        <li>You can view the results by clicking on the <b>Previous</b> and <b>Next</b> buttons.</li>
+        <li>To add a status or test result, click on <b>Accept</b> button to mark the data as passed. Otherwise, click on the <b>Reject</b> button.</li>
+        <li>Once you are done, download the CSV file.</li>
+      </ol>
+      
+    </div>
+  )
+}
+const Navigation = (props) => {
+  return (
+    <div className="navigation">
+      <button onClick={props.onPrev}>{'\u2B05'} Previous</button>
+      <button onClick={props.onModalClose}>Close</button>
+      <button onClick={props.onNext}>Next {'\u27A1'}</button>    
+    </div>
+  )
+}
+
+const Download = (props) => {
+  return (
+    <div className="download">
+      <button id="downloadBtn" onClick={props.onDownload} value="download">Download</button>
+    </div>
+  )
+}
+
+const Preview = (props) => {
+  return (
+    <div id="preview-all">
+      <div className="column">
+        <p>
+          <strong>PAGE {props.activeData + 1}:</strong>
+          <a href={props.oldUrl}> {props.oldUrl}</a>
+        </p>
+        <iframe title="preview-0" className="preview" src={'https://image.thum.io/get/width/1024/fullpage/allowJPG' + props.oldUrl} width="100%" height="500" />
+      </div>
+      <div className="column">
+        <p>
+          <strong>PAGE {props.activeData + 1}:</strong>
+          <a href={props.newUrl}> {props.newUrl}</a>
+        </p>
+        <iframe title="preview-1" className="preview" src={'https://image.thum.io/get/width/1024/fullpage/allowJPG' + props.newUrl} width="100%" height="500" />
+      </div>
+    </div>
+  )
+}
+
+const Modal = (props) => {
+  return (
```

Comment:
**Suggestion (non-blocking)**: One thing that I've found really helpful is asking what state can be owned by each subcomponent!

Right now, all the state, (and thus handlers) is hoisted all the way up to the top `App` component. That works fine, but it can make the `App` component a bit bloated, and require subcomponents to take in a lot of props.

Generally speaking, you only need to "hoist up" state that is used across several parallel components.

For example, the main piece of shared state that stands out to me is the `parsedData` and maybe `newData` -- this is what is shared across the file loading components and the `Modal`. 

You could then move the other pieces of state down closer to where they are used! That also means the handlers that interact with those pieces of state could also move down.

For example...

- `Modal` could own `newData`, `activeData`, and the associated handlers
- You could make a new component that wraps the filePicker that owns `selectedFile` and `isFilePicked`

Reply:
Breaking them down into smaller components was harder than I thought! 

In this commit, [f757a15](https://github.com/Automattic/compare-snapshot-tool/pull/1/commits/f757a15993b80ccd0130adb0a080f956d90568b1), I have created a `components` folder and separated the following:

1. Site preview
2. Navigation
3. Modal
4. Download
5. Instructions 

In case we need to expand or add more functionalities!